File: week01/pset1/mario.c
/**
 * mario.c
 * this is part of CS50X pset1
 *
 * Otto Brut
 * fazzolini@gmail.com
 *
 * version: 1.01
 *
 * Builds a mario-style pyramid.
 */

#include <cs50.h>
#include <stdio.h>

// prototype
void printLine(int line_width, int n_blocks_at_end);

int main(void)
{

    // dealing with input first
    int height;
    do
    {
        printf("Height: ");
        height = GetInt();
    }
    while ((height < 0) || (height > 23));
    
    // check for 0 border case
    if (height == 0)
    {
        // do nothing
        return 0;
    }
    
    // building the pyramid
    int width = height + 1;
    
    for (int i = 1; i <= height; i++)
    {
        printLine(width, i + 1);
    }
    
}

/*
 * this function prints a line of spaces
 * of a given line_width
 * with n_blocks_at_end 
 * pound signs at the end
 */
void printLine(int line_width, int n_blocks_at_end)
{

    int n_spaces = line_width - n_blocks_at_end;
    for (int i = 1; i <= line_width; i++)
    {
        if (i <= n_spaces)
        {
            printf(" ");
        } else
        {
            printf("#");
        }
    }
    printf("\n");

}

File: week08/pset6/mario.py
import cs50

def print_pyramid(h):
    if h == 0:
        return 1
    else:
        # width = height + 1
        w = h + 1
        # iterate through each row
        for r in range(h):
            # in each row, print (h - r) spaces
            for s in range(h - r):
                print(" ", end="")
            # then print (r + 1) # signs
            for p in range(r + 2):
                print("#", end="")
            # new line
            print("\n", end="")
    
def main():
    while True:
        print("Height: ", end="")
        h = cs50.get_int()
        if h >= 0 and h <= 23:
            break
    
    print_pyramid(h)
    
    return 0
    
if __name__ == "__main__":
    main()